packages/blade/src/components/Skeleton/PulseAnimation.web.tsx

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable @typescript-eslint/restrict-plus-operands */
+import type { DefaultTheme, Keyframes } from 'styled-components';
+import styled, { css, keyframes } from 'styled-components';
+import BaseBox from '~components/Box/BaseBox';
+import { castWebType, makeMotionTime } from '~utils';
+
+const pulseKeyframes = ({
+  theme,
+  contrast,
+}: {
+  theme: DefaultTheme;
+  contrast: 'low' | 'high';
+}): Keyframes => keyframes`
+  0% {
+    background-color: ${theme.colors.brand.gray.a100[`${contrast}Contrast`]};
+  }
+  25% {
+    background-color: ${theme.colors.brand.gray.a100[`${contrast}Contrast`]};
+  }
+  100% {
+    background-color: ${theme.colors.brand.gray.a50[`${contrast}Contrast`]};
+  }
+`;
+
+const fadeInKeyframes = (): Keyframes => keyframes`
+  0% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
+`;
+
+const PulseAnimation = styled(BaseBox)<{ contrast: 'low' | 'high' }>(({ theme, contrast }) => {
+  // We need to delay the animation in between keyframes
+  // Since we also offset the animation to have 300ms delay in 25% keyframe
+  // https://css-tricks.com/css-keyframe-animation-delay-iterations/
+  const delay = 300;
+  const duration = castWebType(makeMotionTime(theme.motion.duration['2xgentle'] + delay));
```

Comment:
Why + delay?

Reply:
Basically the pulsating animation has a 300ms waiting period between keyframes thus you can see I have a 25% keyframe which keeps the bg color as is (simulating a 300ms delay between the starting and ending frame). 

Coming to this delay variable, it's actually not a delay but a durationOffset (i'll rename it) - since we want 300ms waiting period between the frames that means now the animation duration is 600ms but we want it to be 900ms exact, thus we add the offset of 300ms back in. 

<img width="645" alt="image" src="https://github.com/razorpay/blade/assets/35374649/03ceb160-893a-465c-913d-2b146c34aae7">